packages/autorest.python/autorest/codegen/models/constant_type.py

```diff
@@ -75,7 +76,17 @@ def docstring_type(self, **kwargs: Any) -> str:
         return self.value_type.docstring_type(**kwargs)
 
     def type_annotation(self, **kwargs: Any) -> str:
-        return self.value_type.type_annotation(**kwargs)
+        return (
+            f"Literal[{self.get_declaration()}]"
+            if self._is_literal
+            else self.value_type.type_annotation(**kwargs)
+        )
+
+    @property
+    def _is_literal(self) -> bool:
+        return isinstance(
```

Comment:
what are the constant types that can't be literals? we can only do constants of primitives really, so I'm not sure which primitive type would fail this

Reply:
Referring to https://mypy.readthedocs.io/en/stable/literal_types.html#parameterizing-literals, "Literal types may contain one or more literal bools, ints, strs, bytes, and enum values.".
I filter out Float/AnyType/DatetimeType/TimeType/DateType/DurationType in our primitive types.

Reply:
I see, yeah we should add this code back in, thanks!